test/appium/tests/base_test_case.py

```diff
@@ -396,8 +390,16 @@ def teardown_class(cls):
         from tests.conftest import sauce
         requests_session = requests.Session()
         requests_session.auth = (sauce_username, sauce_access_key)
+        if test_suite_data.tests[0].testruns[-1].error and 'setup failed' in test_suite_data.tests[0].testruns[
+            -1].error:
+            group_setup_failed = True
+        else:
+            group_setup_failed = False
```

Comment:
maybe it is better to have group_setup_failed as False by default and change it if the condition is True?